website/blog/2025-03-14-20-must-try-mongodb-queries-operations-ferretdb.md

```diff
@@ -0,0 +1,463 @@
+---
+slug: 20-must-try-mongodb-queries-operations-ferretdb
+title: '20 must-try MongoDB operations and queries on FerretDB'
+authors: [alex]
+description: >
+  FerretDB just got better with the release of v2, bringing deeper MongoDB compatibility, and enabling more advanced workloads to run complex queries for most use cases. Find out in this post.
+image: /img/blog/mongodb-operations-ferretdb.jpg
+tags: [observability, product, open source]
+---
+
+![20 Must-Try Advanced MongoDB Queries on FerretDB](/img/blog/mongodb-operations-ferretdb.jpg)
+
+FerretDB just got better with [the release of v2](https://blog.ferretdb.io/ferretdb-releases-v2-faster-more-compatible-mongodb-alternative/), bringing deeper MongoDB compatibility, and enabling more advanced workloads to run complex queries for most use cases.
+
+<!--truncate-->
+
+Beyond just basic CRUD operations, most developers and businesses work with large datasets with real-time analytical queries, text and vector indexing, geospatial searches, replication, among other features.
+All of these operations are available on FerretDB – in open source and without any danger of vendor lock-in or proprietary limitations.
+
+In this post, we're diving into 20 advanced MongoDB features and operations you can run on FerretDB.
+We will cover basic queries, authentication, indexes, aggregation operations, geospatial queries, time series collections, and more.
+
+## Prerequisites
+
+- FerretDB v2 instance running on your local machine or server.
+  See the [installation guide](https://docs.ferretdb.io/installation/) for more.
+- `mongosh`
+- `mongoimport`
+
+## Basic queries
+
+If you're new to FerretDB, you might want to start with the basics – [understanding how to set up an instance](https://docs.ferretdb.io/installation/) and how to [run some basic CRUD operations](https://docs.ferretdb.io/usage/).
+You can also check out this [MongoDB CRUD queries guide](https://blog.ferretdb.io/mongodb-crud-operations-with-ferretdb/) before diving into the other sections in this post.
+
+### 1. Insert document with `mongoimport`
+
+Without data, there's nothing to query!
+So we will start by adding some documents to our database instance.
+
+We already prepared a sample dataset to use; it's a collection of books with details like title, authors, genres, publication date, and price.
+See the [books.fixture.json](https://raw.githubusercontent.com/FerretDB/FerretDB/main/website/docs/guides/requests/books.fixture.json) file for the sample data.
```

Comment:
Linking to the file in the `main` branch would break many examples when we modify it